src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/dto/TestCaseDTO.java

```diff
@@ -33,4 +36,45 @@ public TestCaseDTO(String name, String classname, double time) {
     public boolean isSuccessful() {
         return CollectionUtils.isEmpty(errors) && CollectionUtils.isEmpty(failures);
     }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public List<String> getMessage() {
+        return extractMessageFromTestCase().map(List::of).orElse(List.of());
+    }
+
+    private Optional<String> extractMessageFromTestCase() {
+        var hasErrors = !CollectionUtils.isEmpty(errors());
+        var hasFailures = !CollectionUtils.isEmpty(failures());
+        var hasSuccessInfos = !CollectionUtils.isEmpty(successInfos());
```

Comment:
```suggestion
        boolean hasErrors = !CollectionUtils.isEmpty(errors());
        boolean hasFailures = !CollectionUtils.isEmpty(failures());
        boolean hasSuccessInfos = !CollectionUtils.isEmpty(successInfos());
```
I know it was like this before, but IIRC our guidelines mention that `var` should not be used for primitive and/or short types.